python/ray/experimental/state/common.py

```diff
@@ -193,8 +232,29 @@ class State(StateSchema):
     # Returns {"column_a", "column_b"}
     s.columns()
     ```
+
+    In addition, the schema also provides a humanify abstract method to
+    convert the state object into something human readable, ready for printing.
+
+    Subclasses should override this method, providing logic to convert its own fields
+    to something human readable, packaged and returned in a dict.
+
+    Each field that wants to be humanified should include a 'format_fn' key in its
+    metadata dictionary.
     """
 
+    @classmethod
+    def humanify(cls, state: dict) -> dict:
+        """Convert the given state object into something human readable."""
+        for f in fields(cls):
+            if (
+                f.metadata.get("format_fn") is not None
+                and f.name in state
+                and state[f.name] is not None
+            ):
+                state[f.name] = f.metadata["format_fn"](state[f.name])
```

Comment:
i found out an issue which raised error due to some of the data not correct, I think let's do best effort here instead of disallowing users to see the outputs. 

https://github.com/ray-project/ray/issues/35130

Reply:
Just confirming, the issue doesn't have anything to do with this PR? I tried running without the Humanify logic, but still get the incorrect data (year 584556019 for worker_launch_time)

Reply:
> i found out an issue which raised error due to some of the data not correct, I think let's do best effort here instead of disallowing users to see the outputs.
> 
> #35130

I am new to this;
By "best effort," what did you have in mind? If there is an error in the format_fn, we just leave the data as is. Where do we disallow users to see the outputs?

Reply:
> nit: let' also add a try...except around this and logger errors instead of failing:
> 
> ```
>                 try:
>                     state[f.name] = f.metadata["format_fn"](state[f.name])
>                 except Exception as e:
>                     logger.error(f"Failed to format {f.name}:{state[f.name]} with {e}")
> ```

Also, added a null check to pass state_log unit test.

Reply:
> Just confirming, the issue doesn't have anything to do with this PR? I tried running without the Humanify logic, but still get the incorrect data (year 584556019 for worker_launch_time)

Yes, it's not caused by this PR.